src/Services/CampaignDraftService.php

```diff
@@ -0,0 +1,253 @@
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:findByCampaignDraftId() failed",
+                $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * create a new fresh CampaignDraft
+     * @param Campaigndraft $campaignDraft
+     */
+    public function create(CampaignDraft $campaignDraft)
+    {
+        $response = $this->mailjet->post(Resources::$Campaigndraft,
+            ['body' => $campaignDraft->format()]);
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:create() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Update one specific campaigndraft resource
+     * @param int $id
+     * @param Campaigndraft $campaignDraft
+     */
+    public function update($CampaignId, CampaignDraft $campaignDraft)
+    {
+        $response = $this->mailjet->put(Resources::$Campaigndraft,
+            ['id' => $CampaignId, 'body' => $campaignDraft->format()]);
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:update() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Return the text and html contents of the campaigndraft
+     * @param string $id
+     * @return array
+     */
+    public function getDetailContent($id)
+    {
+        $response = $this->mailjet->get(Resources::$CampaigndraftDetailcontent,
+            ['id' => $id]);
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:getDetailContent failed",
+                $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Creates the content of a  campaigndraft
+     * @param string $id
+     * @return array
+     */
+    public function createDetailContent($id, $contentData)
+    {
+        $response = $this->mailjet->post(Resources::$CampaigndraftDetailcontent,
+            ['id' => $id, 'body' => $contentData]);
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:createDetailContent failed",
+                $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Return the date of the scheduled sending of the campaigndraft
+     * @param string Campaign $id
+     * @return array
+     */
+    public function getSchedule($CampaignId)
+    {
+        $response = $this->mailjet->get(Resources::$CampaigndraftSchedule,
+            ['id' => $CampaignId]);
+        if (!$response->success()) {
+            $this->throwError("CampaignDraftManager:getSchedule failed",
+                $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Schedule when the campaigndraft will be sent
+     * @param string Campaign $id
+     * @param string Schedule $date
+     * @return array
+     */
+    public function scheduleCampaign($CampaignId, $date)
```

Comment:
Not clear the date parameter should be formatted according to https://dev.mailjet.com/guides/#schedule-a-campaign